kgo.js

```diff
@@ -1,41 +1,27 @@
 'use strict';
 
-const fs = require('fs');
-const path = require('path');
+/* eslint func-call-spacing: "off", no-unexpected-multiline: "off" */
 
-const foreign = require('foreign');
-const kgo = require('kgo');
-const R = require('ramda');
-const S = require('sanctuary');
+const foreign       = require('foreign');
+const kgo           = require('kgo');
+const S             = require('sanctuary');
 
-
-//    join :: String -> String -> String
-const join = R.curryN(2, path.join);
-
-//    data Text = Buffer | String
-//    readFile ::
-//      String -> String -> ((Error?, Text?) -> Undefined) -> Undefined
-const readFile = R.curry((encoding, filename, callback) => {
-  fs.readFile(filename, {encoding: encoding}, callback);
-});
+const exit0         = require('./common/exit0');
+const exit1         = require('./common/exit1');
+const join          = require('./common/join');
+const readFile      = require('./common/read-file-callback');
 
 
 const main = () => {
-  const dir = process.argv[2];
+  const path = join(process.argv[2]);
 
   kgo
-  ('index', readFile('utf8', join(dir, 'index.txt')))
-  ('filePaths', ['index'], kgo.sync(S.compose(R.map(join(dir)), S.lines)))
-  ('files', ['filePaths'], R.partial(foreign.parallel, [readFile('utf8')]))
-  ('concated', ['files'], kgo.sync(R.join('')))
-  (['concated'], data => {
-    process.stdout.write(data);
-    process.exit(0);
-  })
-  (['*'], err => {
-    process.stderr.write(String(err) + '\n');
-    process.exit(1);
-  });
+  ('index', readFile(path('index.txt')))
+  ('filePaths', ['index'], kgo.sync(S.compose(S.map(path), S.lines)))
+  ('files', ['filePaths'], S.curry3(foreign.parallel, readFile))
+  ('concated', ['files'], kgo.sync(S.joinWith('')))
+  (['concated'], exit0)
+  (['*'], exit1);
 };
 
 if (process.mainModule.filename === __filename) main();
```

Comment:
Honestly I'm not advocating for kgo's usage these days, but rather suggest people look at righto. I'd be happy to drop this case.